spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java

```diff
@@ -54,7 +54,7 @@ class SparkFilesScan extends SparkScan {
   @Override
   protected List<CombinedScanTask> taskGroups() {
     if (tasks == null) {
-      FileScanTaskSetManager taskSetManager = FileScanTaskSetManager.get();
```

Comment:
Can we remove this class and `SparkFilesScanBuilder` completely?

Reply:
Shall we also drop `FileScanTaskSetManager`?

Reply:
Oh, wait, 1.2 is not out yet.

Reply:
If we plan to merge before 1.2, it is not safe to change `SparkFilesScan`.

Reply:
I'm ok to wait until 1.2 is merged

Reply:
I agree that this PR should only be merged once 1.2.0 is officially out. Also would be great to delete `FileScanTaskSetManager` as part of this PR

Reply:
Removed in latest commit